### PR TITLE
Add manual mode to noise reduction + add noise reduction threshold

### PR DESCRIPTION
**Pull request by digisomni**
Generally, it did a damn great job of filtering out my keystrokes though. Users that find a good number for the threshold I think will have a good time with this. Pretty much all voice applications (ever) have manual noise gates. I always use manual mode, it helps for tuning what I want to convey as an experience instead of leaving it to algorithms.

image

### CODE REVIEW

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-PR1007-ffd6f79_5a400077.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-PR1007-ffd6f79.tar_5a400077.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-PR1007-ffd6f79.tar_5a400077.Z

**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-PR1007-ffd6f79_5a400077.dmg

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-PR1007-ffd6f79_5a400077.exe

**build (ubuntu-18.04, android)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-PR1007-ffd6f79_5a400077.apk
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-Quest-Alpha-PR1007-ffd6f79_5a400077.apk

**Comment by digisomni** [comment]
This is ready for review now.

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-PR1007-6bb90c2.tar_97fdd372.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-PR1007-6bb90c2_97fdd372.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-PR1007-6bb90c2.tar_97fdd372.Z

**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-PR1007-6bb90c2_97fdd372.dmg

**build (ubuntu-18.04, android)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-Quest-Alpha-PR1007-6bb90c2_97fdd372.apk
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-PR1007-6bb90c2_97fdd372.apk

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-PR1007-6bb90c2_97fdd372.exe

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-PR1007-d429b6e_799590a0.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-PR1007-d429b6e.tar_799590a0.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-PR1007-d429b6e.tar_799590a0.Z

**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-PR1007-d429b6e_799590a0.dmg

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-PR1007-d429b6e_799590a0.exe

**build (ubuntu-18.04, android)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-Quest-Alpha-PR1007-d429b6e_799590a0.apk
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-PR1007-d429b6e_799590a0.apk

**Comment by SilverfishVR** [comment]
I seems to work, but I could not find anyone online on master so I did not actually talk to anyone 😄

**Comment by digisomni** [comment]
> I seems to work, but I could not find anyone online on master so I did not actually talk to anyone 😄

I actually talk to myself, no joke. I'd just listen on my other client to see at what points I'd hear me. :)

**Comment by ctrlaltdavid** [comment]
UI suggestion: Change "Noise Reduction: Manual/Automatic" to "Manual Noise Reduction".
This would make it clearer that the toggle enables manual noise reduction?

**Comment by digisomni** [comment]
Currently it's a switch that controls the value of "noiseValueAutomatic" , but I could reverse it and rename it probably, yeah.

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-PR1007-0dca3d9_88b9ebb1.dmg

**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-PR1007-0dca3d9.tar_88b9ebb1.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-PR1007-0dca3d9.tar_88b9ebb1.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-PR1007-0dca3d9_88b9ebb1.sh

**build (ubuntu-18.04, android)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-PR1007-0dca3d9_88b9ebb1.apk
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-Quest-Alpha-PR1007-0dca3d9_88b9ebb1.apk

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-PR1007-0dca3d9_88b9ebb1.exe

File: interface/resources/qml/hifi/audio/Audio.qml [hunk 1]
```diff
@@ -104,6 +105,11 @@ Rectangle {
             AudioScriptingInterface.setSystemInjectorGain(sliderValue);
         }
     }
+    function updateNoiseReductionThresholdFromQML(sliderValue) {
+        if (AudioScriptingInterface.getNoiseReductionThreshold() != sliderValue) {
```
Comment (ctrlaltdavid) [inline interface/resources/qml/hifi/audio/Audio.qml:109]:
```suggestion
        if (AudioScriptingInterface.getNoiseReductionThreshold() !== sliderValue) {
```

File: interface/resources/qml/hifi/audio/Audio.qml [hunk 2]
```diff
@@ -519,13 +492,264 @@ Rectangle {
             anchors.top: systemInjectorGainContainer.bottom;
             anchors.topMargin: 10;
         }
+        
+        Item {
+            id: noiseReductionHeader
+            x: margins.paddings;
+            width: parent.width - margins.paddings*2;
```
Comment (ctrlaltdavid) [inline interface/resources/qml/hifi/audio/Audio.qml:499]:
```suggestion
            width: parent.width - margins.paddings * 2;
```

File: interface/resources/qml/hifi/audio/Audio.qml [hunk 3]
```diff
@@ -519,13 +492,264 @@ Rectangle {
...
+            height: 36;
+            anchors.top: secondSeparator.bottom;
+            anchors.topMargin: 10;
+
+            HiFiGlyphs {
+                width: margins.sizeCheckBox;
+                text: hifi.glyphs.mic;
+                color: hifi.colors.white;
+                anchors.left: parent.left;
+                anchors.leftMargin: -size/4; //the glyph has empty space at left about 25%
```
Comment (ctrlaltdavid) [inline interface/resources/qml/hifi/audio/Audio.qml:509]:
```suggestion
                anchors.leftMargin: -size / 4; // The glyph has empty space at left about 25%
```

File: interface/resources/qml/hifi/audio/Audio.qml [hunk 4]
```diff
@@ -519,13 +492,264 @@ Rectangle {
...
+                    anchors.top: noiseReductionSwitchContainer.top
+                    anchors.topMargin: 24
+                    anchors.left: parent.left
+                    labelTextOn: "Echo Cancellation";
+                    labelTextSize: 16;
+                    backgroundOnColor: "#E3E3E3";
+                    checked: AudioScriptingInterface.acousticEchoCancellation;
+                    onCheckedChanged: {
+                        AudioScriptingInterface.acousticEchoCancellation = checked;
+                        checked = Qt.binding(function() { return AudioScriptingInterface.acousticEchoCancellation; });
```
Comment (ctrlaltdavid) [inline interface/resources/qml/hifi/audio/Audio.qml:555]:
```suggestion
                        checked = Qt.binding(function () { return AudioScriptingInterface.acousticEchoCancellation; });
```

File: interface/resources/qml/hifi/audio/Audio.qml [hunk 5]
```diff
@@ -519,13 +492,264 @@ Rectangle {
...
+                    anchors.top: acousticEchoCancellationSwitch.bottom;
+                    anchors.topMargin: 24
+                    anchors.left: parent.left
+                    labelTextOn: "Noise Reduction";
+                    labelTextSize: 16;
+                    backgroundOnColor: "#E3E3E3";
+                    checked: AudioScriptingInterface.noiseReduction;
+                    onCheckedChanged: {
+                        AudioScriptingInterface.noiseReduction = checked;
+                        checked = Qt.binding(function() { return AudioScriptingInterface.noiseReduction; }); // restore binding
```
Comment (ctrlaltdavid) [inline interface/resources/qml/hifi/audio/Audio.qml:572]:
```suggestion
                        checked = Qt.binding(function () { return AudioScriptingInterface.noiseReduction; }); // restore binding
```

File: interface/resources/qml/hifi/audio/Audio.qml [hunk 6]
```diff
@@ -519,13 +492,264 @@ Rectangle {
...
+                    height: root.switchHeight;
+                    switchWidth: root.switchWidth;
+                    anchors.top: noiseReductionSwitch.bottom;
+                    anchors.topMargin: 24;
+                    anchors.left: parent.left;
+                    labelTextOn: "Manual Noise Reduction";
+                    labelTextSize: 16;
+                    backgroundOnColor: "#E3E3E3";
+                    checked: !AudioScriptingInterface.noiseReductionAutomatic;
+                    visible: AudioScriptingInterface.noiseReduction === true;
```
Comment (ctrlaltdavid) [inline interface/resources/qml/hifi/audio/Audio.qml:587]:
```suggestion
                    visible: AudioScriptingInterface.noiseReduction;
```

File: interface/resources/qml/hifi/audio/Audio.qml [hunk 7]
```diff
@@ -519,13 +492,264 @@ Rectangle {
...
+                    onCheckedChanged: {
+                        AudioScriptingInterface.noiseReductionAutomatic = !checked;
+                        checked = Qt.binding(function() { return !AudioScriptingInterface.noiseReductionAutomatic; }); // restore binding
```
Comment (ctrlaltdavid) [inline interface/resources/qml/hifi/audio/Audio.qml:590]:
```suggestion
                        checked = Qt.binding(function () { return !AudioScriptingInterface.noiseReductionAutomatic; }); // restore binding
```

File: interface/resources/qml/hifi/audio/Audio.qml [hunk 8]
```diff
@@ -519,13 +492,264 @@ Rectangle {
...
+                }
+            }
+        }
+        
+        Item {
+            id: noiseReductionThresholdContainer
+            x: margins.paddings;
+            anchors.top: noiseReductionSwitches.bottom;
+            anchors.topMargin: 16;
+            width: parent.width - margins.paddings*2;
```
Comment (ctrlaltdavid) [inline interface/resources/qml/hifi/audio/Audio.qml:601]:
```suggestion
            width: parent.width - margins.paddings * 2;
```

File: interface/resources/qml/hifi/audio/Audio.qml [hunk 9]
```diff
@@ -519,13 +492,264 @@ Rectangle {
...
+            height: avatarGainSliderTextMetrics.height + 10;
+            visible: AudioScriptingInterface.noiseReduction === true && AudioScriptingInterface.noiseReductionAutomatic !== true;
```
Comment (ctrlaltdavid) [inline interface/resources/qml/hifi/audio/Audio.qml:603]:
```suggestion
            visible: AudioScriptingInterface.noiseReduction && !AudioScriptingInterface.noiseReductionAutomatic;
```

File: interface/resources/qml/hifi/audio/Audio.qml [hunk 10]
```diff
@@ -519,13 +492,264 @@ Rectangle {
...
+                    }
+                }
+
+                MouseArea {
+                    anchors.fill: parent
+                    onWheel: {
+                        // Do nothing.
+                    }
+                    onDoubleClicked: {
+                        noiseReductionThresholdSlider.value = 0.0
```
Comment (ctrlaltdavid) [inline interface/resources/qml/hifi/audio/Audio.qml:629]:
```suggestion
                        noiseReductionThresholdSlider.value = 0.0;
```

File: interface/resources/qml/hifi/audio/Audio.qml [hunk 11]
```diff
@@ -519,13 +492,264 @@ Rectangle {
...
+                    }
+                    onPressed: {
+                        // Pass through to Slider
+                        mouse.accepted = false
```
Comment (ctrlaltdavid) [inline interface/resources/qml/hifi/audio/Audio.qml:633]:
```suggestion
                        mouse.accepted = false;
```

File: interface/resources/qml/hifi/audio/Audio.qml [hunk 12]
```diff
@@ -519,13 +492,264 @@ Rectangle {
...
+                    }
+                    onReleased: {
+                        // the above mouse.accepted seems to make this
+                        // never get called, nonetheless...
+                        mouse.accepted = false
```
Comment (ctrlaltdavid) [inline interface/resources/qml/hifi/audio/Audio.qml:638]:
```suggestion
                        mouse.accepted = false;
```

File: interface/resources/qml/hifi/audio/Audio.qml [hunk 13]
```diff
@@ -519,13 +492,264 @@ Rectangle {
...
+                    height: parent.height;
+
+                    anchors { fill: parent }
+
+                    visible: !status.visible;
+
+                    Component.onCompleted: {
+                        AudioScriptingInterface.noiseGateOpened.connect(function() { noiseBar.gated = false; });
+                        AudioScriptingInterface.noiseGateClosed.connect(function() { noiseBar.gated = true; });
+                        AudioScriptingInterface.inputLevelChanged.connect(function() { noiseBar.level = AudioScriptingInterface.inputLevel; });
```
Comment (ctrlaltdavid) [inline interface/resources/qml/hifi/audio/Audio.qml:698]:
```suggestion
                        AudioScriptingInterface.noiseGateOpened.connect(function () { noiseBar.gated = false; });
                        AudioScriptingInterface.noiseGateClosed.connect(function () { noiseBar.gated = true; });
                        AudioScriptingInterface.inputLevelChanged.connect(function () { noiseBar.level = AudioScriptingInterface.inputLevel; });
```

File: interface/src/scripting/Audio.h [hunk 14]
```diff
@@ -92,6 +92,8 @@ class Audio : public AudioScriptingInterface, protected ReadWriteLockable {
 
     Q_PROPERTY(bool muted READ isMuted WRITE setMuted NOTIFY mutedChanged)
     Q_PROPERTY(bool noiseReduction READ noiseReductionEnabled WRITE enableNoiseReduction NOTIFY noiseReductionChanged)
+    Q_PROPERTY(bool noiseReductionAutomatic READ noiseReductionAutomatic WRITE enableNoiseReductionAutomatic NOTIFY noiseReductionAutomaticChanged)
+    Q_PROPERTY(float noiseReductionThreshold READ getNoiseReductionThreshold WRITE setNoiseReductionThreshold NOTIFY noiseReductionThresholdChanged)
```
Comment (ctrlaltdavid) [inline interface/src/scripting/Audio.h:96]: JSdoc needs to be added for these new properties.

File: interface/src/scripting/Audio.h [hunk 15]
```diff
@@ -270,6 +273,20 @@ class Audio : public AudioScriptingInterface, protected ReadWriteLockable {
      * @returns {number} The injector gain (dB) in the client.
     */
     Q_INVOKABLE float getSystemInjectorGain();
+    
+    /**jsdoc
+     * Sets the noise gate threshold before your mic audio is transmitted. (Applies only if Audio.noiseReductionAutomatic is off.)
```
Comment (ctrlaltdavid) [inline interface/src/scripting/Audio.h:278]:
```suggestion
     * Sets the noise gate threshold before your mic audio is transmitted. (Applies only if <code>Audio.noiseReductionAutomatic</code> is <code>false</code>.)
```

File: interface/src/scripting/Audio.h [hunk 16]
```diff
@@ -398,6 +415,24 @@ class Audio : public AudioScriptingInterface, protected ReadWriteLockable {
      * @returns {Signal}
      */
     void noiseReductionChanged(bool isEnabled);
+    
+    /**jsdoc
+     * Triggered when the audio input noise reduction mode is changed.
+     * @function Audio.noiseReductionAutomaticChanged
+     * @param {boolean} isEnabled - <code>true</code> if audio input noise reduction automatic mode is enabled, otherwise <code>false</code>. 
+     *     This means the mode is set to 'manual'.
```
Comment (ctrlaltdavid) [inline interface/src/scripting/Audio.h:423]:
```suggestion
     * @param {boolean} isEnabled - <code>true</code> if audio input noise reduction automatic mode is enabled, <code>false</code> if in manual mode.
```

Comment (ctrlaltdavid) [inline interface/src/scripting/Audio.h:423]:
```suggestion
     * @param {boolean} isEnabled - <code>true</code> if audio input noise reduction automatic mode is enabled, <code>false</code> if in manual mode.
```

File: interface/src/scripting/Audio.h [hunk 17]
```diff
@@ -398,6 +415,24 @@ class Audio : public AudioScriptingInterface, protected ReadWriteLockable {
...
+     * @returns {Signal}
+     */
+    void noiseReductionAutomaticChanged(bool isEnabled);
+    
+    /**jsdoc
+     * Triggered when audio input noise reduction threshold is changed.
```
Comment (ctrlaltdavid) [inline interface/src/scripting/Audio.h:429]:
```suggestion
     * Triggered when the audio input noise reduction threshold is changed.
```

**Review by ctrlaltdavid — CHANGES_REQUESTED** [review]

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-PR1007-77dd062.tar_0a6e220c.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-PR1007-77dd062_0a6e220c.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-PR1007-77dd062.tar_0a6e220c.Z

**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-PR1007-77dd062_0a6e220c.dmg

**build (ubuntu-18.04, android)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-PR1007-77dd062_0a6e220c.apk
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-Quest-Alpha-PR1007-77dd062_0a6e220c.apk

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-PR1007-77dd062_0a6e220c.exe

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-PR1007-48e03a5_83c8b71f.dmg

**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-PR1007-48e03a5.tar_83c8b71f.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-PR1007-48e03a5_83c8b71f.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-PR1007-48e03a5.tar_83c8b71f.Z

**build (ubuntu-18.04, android)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-Quest-Alpha-PR1007-48e03a5_83c8b71f.apk
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-PR1007-48e03a5_83c8b71f.apk

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-PR1007-48e03a5_83c8b71f.exe

File: interface/src/scripting/Audio.h [hunk 18]
```diff
@@ -50,6 +50,11 @@ class Audio : public AudioScriptingInterface, protected ReadWriteLockable {
      * @property {boolean} noiseReduction - <code>true</code> if noise reduction is enabled, otherwise <code>false</code>. When
      *     enabled, the input audio signal is blocked (fully attenuated) when it falls below an adaptive threshold set just
      *     above the noise floor.
+     * @property {boolean} noiseReductionAutomatic - <code>true</code> if audio input noise reduction automatic mode is enabled, 
+     *     <code>false</code> if in manual mode. Manual mode will allow you to use <code>Audio.noiseReductionAutomatic</code>
```
Comment (ctrlaltdavid) [inline interface/src/scripting/Audio.h:54]:
```suggestion
     *     <code>false</code> if in manual mode. Manual mode allows you to use <code>Audio.noiseReductionThreshold</code>
```

**Review by ctrlaltdavid — CHANGES_REQUESTED** [review]

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-PR1007-49b8645_61defe74.dmg

**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-PR1007-49b8645_61defe74.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-PR1007-49b8645.tar_61defe74.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-PR1007-49b8645.tar_61defe74.Z

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-PR1007-49b8645_61defe74.exe

**build (ubuntu-18.04, android)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-Quest-Alpha-PR1007-49b8645_61defe74.apk
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-PR1007-49b8645_61defe74.apk

**Review by ctrlaltdavid — APPROVED** [review]

**Comment by Aitolda** [comment]
Threshold seems to do what it's supposed to do.. HOWEVER... the "test your mic" audio feed seems to be pre-noise reduction so this is probably not the most ideal place in the signal flow.  Ideally the feed played while you "test your mic| would be after noise reduction.

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-PR1007-d8cdc8e-_520d21b7.dmg

**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-PR1007-d8cdc8e-.tar_520d21b7.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-PR1007-d8cdc8e-.tar_520d21b7.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-PR1007-d8cdc8e-_520d21b7.sh

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-PR1007-d8cdc8e-_520d21b7.exe

**build (ubuntu-18.04, android)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-PR1007-d8cdc8e_520d21b7.apk
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-Quest-Alpha-PR1007-d8cdc8e_520d21b7.apk

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-PR1007-b87126d-_bc2b3457.dmg

**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-PR1007-b87126d-.tar_bc2b3457.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-PR1007-b87126d-.tar_bc2b3457.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-PR1007-b87126d-_bc2b3457.sh

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-PR1007-b87126d-_bc2b3457.exe

**build (ubuntu-18.04, android)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-PR1007-b87126d_bc2b3457.apk
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-Quest-Alpha-PR1007-b87126d_bc2b3457.apk

**Comment by Aitolda** [comment]
Mic listen now working post-noise reduction.  Seems to be working as intended.

**Review by ctrlaltdavid — APPROVED** [review]

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-PR1007-2f33bcd-_809bd936.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-PR1007-2f33bcd-.tar_809bd936.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-PR1007-2f33bcd-.tar_809bd936.gz

**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-PR1007-2f33bcd-_809bd936.dmg

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-PR1007-2f33bcd-_809bd936.exe

**build (ubuntu-18.04, android)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-Quest-Alpha-PR1007-2f33bcd_809bd936.apk
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1007/Vircadia-PR1007-2f33bcd_809bd936.apk

**Comment by Aitolda** [comment]
Update tested. Manual noise reduction no longer default (starts automatic). Otherwise continues to function as before.

**Pull request merged**